network-libp2p/src/connection_pool/behaviour.rs

```diff
@@ -98,6 +99,12 @@ struct ConnectionState<T> {
     connected: BTreeMap<T, Option<Services>>,
     /// Set of connection IDs marked as banned.
     banned: BTreeSet<T>,
+    /// The first peer that can be unbanned when its ban expiration is reached.
+    next_unban_peer: Option<(T, Instant)>,
+    /// List of subsequent peer unban deadlines in ascending order.
+    subsequent_unban_deadlines: VecDeque<(T, Instant)>,
```

Comment:
```suggestion
    /// List of peer unban deadlines in ascending order.
    unban_deadlines: VecDeque<(T, Instant)>,
```
The extra `Option` seems superfluous to me, it's acting as the first element of the `subsequent_unban_deadlines`.